core/src/test/java/tech/tablesaw/io/csv/CsvReaderTest.java

```diff
@@ -173,6 +173,64 @@ public void testDataTypeDetection() throws IOException {
     assertArrayEquals(bus_types, columnTypes);
   }
 
+  @Test
+  public void testNumberDetectionRetainZeroDecimals() throws IOException {
+    Reader reader = new FileReader("../data/immunization.csv");
+    CsvReadOptions options =
+        CsvReadOptions.builder(reader).header(true).sample(false).ignoreZeroDecimal(true).build();
+
+    // Column index 3 and 7 contain values with none to 3 zero values as suffix
+    // Should map to type INTEGER when ignoreZeroDecimal = true
+    ColumnType[] columnTypes = new CsvReader().detectColumnTypes(reader, options);
+    assertEquals(INTEGER, columnTypes[3]);
+    assertEquals(INTEGER, columnTypes[7]);
+  }
+
+  @Test
+  public void testNumberColumnTypeRetainZeroDecimals() throws IOException {
+    CsvReadOptions options =
+        CsvReadOptions.builder("../data/immunization.csv")
+            .header(true)
+            .sample(false)
+            .ignoreZeroDecimal(true)
+            .build();
+
+    // Column index 3 and 7 contain values with none to 3 zero values as suffix
+    // Should map to type INTEGER when ignoreZeroDecimal = true
+    Table table = Table.read().csv(options);
+    assertEquals(INTEGER, table.column(3).type());
+    assertEquals(INTEGER, table.column(7).type());
+  }
+
+  @Test
+  public void testNumberDetectionIgnoreZeroDecimals() throws IOException {
+    Reader reader = new FileReader("../data/immunization.csv");
+    CsvReadOptions options =
+        CsvReadOptions.builder(reader).header(true).sample(false).ignoreZeroDecimal(false).build();
```

Comment:
The auto-formatter renders this on one line, originally I had one line per operation.